src/Components/Server/src/ComponentHub.cs

```diff
@@ -275,12 +277,27 @@ private async ValueTask<CircuitHost> GetActiveCircuitAsync([CallerMemberName] st
             return circuitHost;
         }
 
-        private CircuitHost GetCircuit()
+        // We wrap the `TryDeserializeComponentDescriptorCollection` call in a virtual method here
+        // to support mocking in tests. SignalR does not support having a public method with an out
+        // parameter defined on the hub so we return the desired values in a tuple.
+        public virtual (bool, List<ComponentDescriptor>)  DeserializeComponentDescriptor(string serializedComponentRecords)
```

Comment:
public methods on hubs are public api that clients can call. I don't think you want clients calling these methods, so it should be internal with IVT for tests.